Add TBH support

duc996 opened the pull request:
This pull request adds support for devices from the tbh project (manufactured by archos).

zuckschwerdt commented:
Looks good, thanks! I'll review shortly. Do we have sample files in https://github.com/merbanan/rtl_433_tests ?

duc996 commented:
I have just added a sample file and created the pull request.

duc996 commented:
Sorry for the delay, I have added a sample file just after receiving your message. I have added a comment on github.

----- Original Message -----
From: "Christian W. Zuckschwerdt" <notifications@github.com>
To: "merbanan/rtl_433" <rtl_433@noreply.github.com>
Cc: "duc" <duc_996@gmx.net>, "Author" <author@noreply.github.com>
Sent: Friday, November 15, 2019 10:52:06 AM
Subject: Re: [merbanan/rtl_433] Add TBH support (#1199)

Looks good, thanks! I'll review shortly. Do we have sample files in https://github.com/merbanan/rtl_433_tests ?

-- 
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub:
https://github.com/merbanan/rtl_433/pull/1199#issuecomment-554292066

merbanan commented on src/devices/tbh.c:
```diff
@@ -0,0 +1,275 @@
+  .modulation = FSK_PULSE_PCM,
+  .short_width = 212,
+  .long_width = 212,
+  .reset_limit = 217088,
```
Reset limit is high.

merbanan commented on src/devices/tbh.c:
```diff
@@ -0,0 +1,275 @@
+      fprintf (stderr, "got CRC %04x\n", crc);
+
+    if ((frame[len + 1] << 8 | frame[len + 2]) != crc) {
+      if (decoder->verbose) {
```
MIC maybe should be added to the output.

zuckschwerdt commented:
And Indent is 2 instead of 4. I'll just merge shortly and the clean it up.

zuckschwerdt commented:
Clean up in 8ed6eaf. Note that the single sample we have shows a truncated preamble.
